src/main/java/software/amazon/smithy/lsp/ext/DependencyDownloader.java

```diff
@@ -0,0 +1,82 @@
+package software.amazon.smithy.lsp.ext;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+
+import coursierapi.Dependency;
+import coursierapi.Fetch;
+import coursierapi.MavenRepository;
+import coursierapi.Repository;
+import coursierapi.error.CoursierError;
+
+public class DependencyDownloader {
+  private List<Repository> repositories;
```

Comment:
remove those mutable fields, just expose a function that takes an instance of SmithyBuildExtensions as an input and outputs whatever you need. 

You can make that function take a Fetch as a parameter, and overload it with a default value for that Fetch.